src/app/games/diam/DiamDummyMinimax.ts

```diff
@@ -0,0 +1,76 @@
+import { Coord } from 'src/app/jscaip/Coord';
+import { LegalityStatus } from 'src/app/jscaip/LegalityStatus';
+import { Minimax } from 'src/app/jscaip/Minimax';
+import { NodeUnheritance } from 'src/app/jscaip/NodeUnheritance';
+import { Player } from 'src/app/jscaip/Player';
+import { GameStatus } from 'src/app/jscaip/Rules';
+import { DiamMove, DiamMoveDrop, DiamMoveShift } from './DiamMove';
+import { DiamPiece } from './DiamPiece';
+import { DiamNode, DiamRules } from './DiamRules';
+import { DiamState } from './DiamState';
+
+export class DiamDummyMinimax extends Minimax<DiamMove, DiamState, LegalityStatus> {
+    public getListMoves(node: DiamNode): DiamMove[] {
+        const state: DiamState = node.gameState;
+        const moves: DiamMove[] = [];
+        const remainingPieces: DiamPiece[] = this.getRemainingPiecesForCurrentPlayer(state);
+        const shiftSources: Coord[] = this.getShiftSources(state);
+        for (let x: number = 0; x < 8; x++) {
+            const height: number = state.getStackHeight(x);
+            for (let y: number = 3; y >= 0; y--) {
+                const piece: DiamPiece = state.getPieceAtXY(x, y);
+                if (piece === DiamPiece.EMPTY) {
+                    // it can be the target for a drop or a shift
+                    for (const shiftSource of shiftSources) {
+                        if (state.getStackHeight(shiftSource.x) - (3 - shiftSource.y) + height <= 4) {
```

Comment:
explain it to me like I'm five :D !
or make it more code-like than quantum-physic style :D !

Reply:
je refactorise suite à l'amélioration du board avec le fromRepresentation, ça devrait être moins cryptique :D